snapcraft/internal/build_providers/_base_provider.py

```diff
@@ -411,7 +411,7 @@ def _get_home_directory(self) -> pathlib.Path:
 
     def _base_has_changed(self, base: str, provider_base: str) -> bool:
         # Make it backwards compatible with instances without project info
-        if base == "core18" and provider_base is None:
+        if base in ("core18", "core20") and provider_base is None:
```

Comment:
this should not be needed, this is transitional logic, newer projects should not expect to see this.

Reply:
Can't remember excatly where i needed this. 

But i think it was on building the core20. 

There i switched the base to core18 and got an error that the base is None. (or so...)